src/code_7bb60_len_41b0.c

```diff
@@ -37,9 +73,37 @@ INCLUDE_ASM(s32, "code_7bb60_len_41b0", update_fall_state);
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E2F60);
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", gravity_use_fall_parms);
+void gravity_use_fall_params(void) {
+    f32* floats = D_800F7B60;
+    PlayerStatus* playerStatus2;
+    PlayerStatus* playerStatus;
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E3100);
+    playerStatus = &gPlayerStatus;
+    playerStatus2 = &gPlayerStatus;
+
+    if (playerStatus->flags & 0x40000) {
+        playerStatus->gravityIntegrator[0] = *floats++ / 12.0f;
+        playerStatus->gravityIntegrator[1] = *floats++ / 12.0f;
+        playerStatus->gravityIntegrator[2] = *floats++ / 12.0f;
+        playerStatus->gravityIntegrator[3] = *floats++ / 12.0f;
+    } else {
+        playerStatus2->gravityIntegrator[0] = *floats++;
+        playerStatus2->gravityIntegrator[1] = *floats++;
+        playerStatus2->gravityIntegrator[2] = *floats++;
+        playerStatus2->gravityIntegrator[3] = *floats++;
+    }
+}
+
+void func_800E3100(void) {
+    PlayerStatus* playerStatus = PLAYER_STATUS;
+
+    if ((playerStatus->actionState != 7) && (playerStatus->actionState != ActionState_BOUNCE)) {
```

Comment:
too many parens here i think

Reply:
also it's probably worth making a #define for ActionState_7 just so its easy to name later when we figure out what it is